File: Novels/packages/5-mobile-ios/StoryReader/Model/Tip.swift
import Foundation

struct Tip: Codable, Identifiable {
    let id: String
    let storyId: String
    let authorId: String?
    let userId: String
    let username: String
    let amount: Int
    let message: String?
    let createdAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id
        case storyId = "story_id"
        case authorId = "author_id"
        case userId = "user_id"
        case username
        case amount
        case message
        case createdAt = "created_at"
    }
}

struct TipStats: Codable {
    let totalTips: Int
    let totalAmount: Int
    let topSupporters: [TipSupporter]
    
    enum CodingKeys: String, CodingKey {
        case totalTips = "total_tips"
        case totalAmount = "total_amount"
        case topSupporters = "top_supporters"
    }
}

struct TipSupporter: Codable, Identifiable {
    let id: String
    let username: String
    let amount: Int
}

File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/Tip.kt
package com.storysphere.storyreader.model

data class Tip(
    val id: String,
    val userId: String,
    val storyId: String,
    val authorId: String,
    val amount: Int, // Amount in points
    val message: String? = null,
    val createdAt: Long = System.currentTimeMillis()
)

data class AuthorTippingStats(
    val authorId: String,
    val totalTips: Int,
    val tipCount: Int,
    val topSupporters: List<TipSupporter>
)

data class TipSupporter(
    val userId: String,
    val username: String,
    val avatar: String? = null,
    val totalAmount: Int,
    val tipCount: Int
)
